packages/flutter_tools/lib/src/test/coverage_collector.dart

```diff
@@ -12,57 +12,107 @@ import '../base/io.dart';
 import '../dart/package_map.dart';
 import '../globals.dart';
 
+/// Class that represents a pending task of coverage data collection.
+/// Instances of this class are obtained when a process starts running code
+/// by calling [CoverageCollector.addTask]. Then, when the code has run to
+/// completion (all the coverage data has been recorded), the task is started
+/// to actually collect the coverage data.
+abstract class CoverageCollectionTask {
```

Comment:
I would just rename _Task to CoverageCollectionTask and make all its members except start() private. That would simplify the code (no need to have this interface).